### PR TITLE
server: apply rules when VR of Domain VPC is recreated

### PR DESCRIPTION
**Pull request by GaOrtiga**
### Description

After re-creating a VR from a domain VPC, ACS will not apply static nat, port forward and load balancer rules to IPs that are not associated to the account that owns the VPC.
An adjustment was made to make sure that these rules are applied for all IPs when performing the following procedures:

* VR restart

* VPC restart

* VPC restart with cleanup

* recreate VR when health-check fails

* recreate VR on failover

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [X] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

In a local lab, I performed the procedures listed in the description and verified that after applying the changes, the IPs owned by accounts different than the VPC owner were having their rules applied accordingly.

### CODE REVIEW

**Review by DaanHoogland — APPROVED** [review]
clgtm, looks like a good cleanup

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Review by GutoVeronezi — APPROVED** [review]
CLGTM

File: server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java [hunk 1]
```diff
@@ -2681,18 +2681,11 @@ protected void finalizeIpAssocForNetwork(final Commands cmds, final VirtualRoute
...
         final List<? extends IpAddress> userIps;
 
-        final Network guestNetwork = _networkDao.findById(guestNetworkId);
-        if (guestNetwork.getGuestType() == GuestType.Shared) {
-            // ignore the account id for the shared network
-            userIps = _networkModel.listPublicIpsAssignedToGuestNtwk(guestNetworkId, null);
-        } else {
-            userIps = _networkModel.listPublicIpsAssignedToGuestNtwk(ownerId, guestNetworkId, null);
-        }
+        userIps = _networkModel.listPublicIpsAssignedToGuestNtwk(guestNetworkId, null);
```
Comment (GutoVeronezi) [inline server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java:2688]:
```suggestion
        final List<? extends IpAddress> userIps = _networkModel.listPublicIpsAssignedToGuestNtwk(guestNetworkId, null);
```

File: server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java [hunk 2]
```diff
@@ -2681,18 +2681,11 @@ protected void finalizeIpAssocForNetwork(final Commands cmds, final VirtualRoute
         }
     }
 
-    protected ArrayList<? extends PublicIpAddress> getPublicIpsToApply(final VirtualRouter router, final Provider provider, final Long guestNetworkId,
+    protected ArrayList<? extends PublicIpAddress> getPublicIpsToApply(final Provider provider, final Long guestNetworkId,
             final com.cloud.network.IpAddress.State... skipInStates) {
-        final long ownerId = router.getAccountId();
         final List<? extends IpAddress> userIps;
```
Comment (GutoVeronezi) [inline server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java:2686]:
```suggestion
```

**Review by GutoVeronezi — COMMENTED** [review]
Just a minor adjustment.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by GaOrtiga** [comment]
@DaanHoogland @GutoVeronezi @shwstppr @JoaoJandre 

Is there any pending concern regarding this or can we proceed with the merge?

**Comment by weizhouapache** [comment]
> @DaanHoogland @GutoVeronezi @shwstppr @JoaoJandre
> 
> Is there any pending concern regarding this or can we proceed with the merge?

@GaOrtiga 
can you ask someone to test it ?

**Comment by sureshanaparti** [comment]
> > @DaanHoogland @GutoVeronezi @shwstppr @JoaoJandre
> > Is there any pending concern regarding this or can we proceed with the merge?
> 
> @GaOrtiga can you ask someone to test it, share the test steps and results ?

Ping @GaOrtiga Any update on the testing and results?

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
Ping @GaOrtiga Any further update?

**Comment by GaOrtiga** [comment]
> Ping @GaOrtiga Any further update?

No updates, just needs testing, I will check if someone can test and post results.

**Comment by weizhouapache** [comment]
@GaOrtiga 
I am trying to test this feature. However, I cannot find any UI/doc change for the domain VPC (#7153 ).
can you please add the UI changes and create a doc PR ?

**Comment by weizhouapache** [comment]
tested ok. cc @sureshanaparti @GaOrtiga

**Review by weizhouapache — APPROVED** [review]

**Comment by GaOrtiga** [comment]
> @GaOrtiga I am trying to test this feature. However, I cannot find any UI/doc change for the domain VPC (#7153 ). can you please add the UI changes and create a doc PR ?

@weizhouapache Thank you for testing. Yes I will create the doc PR.

**Comment by sureshanaparti** [comment]
> > @GaOrtiga I am trying to test this feature. However, I cannot find any UI/doc change for the domain VPC (#7153 ). can you please add the UI changes and create a doc PR ?
> 
> @weizhouapache Thank you for testing. Yes I will create the doc PR.

Hi @GaOrtiga Please update the doc PR reference here once it is ready. Thanks.

**Comment by GaOrtiga** [comment]
@weizhouapache @sureshanaparti 
Link for the doc PR:
https://github.com/apache/cloudstack-documentation/pull/410

**Pull request merged**